### DISCUSSION TITLE
Changing Dahua Doorbell DHI-VTO2202F-P-S2 audio codec

### DISCUSSION THREAD

**Discussion by felipecrs** [blakeblackshear/frigate#2572] [Show and tell]
UPDATE: I'm now using https://github.com/AlexxIT/go2rtc/issues/49#issuecomment-1288196783

---

I was about to give up after writing [this comment here](https://github.com/blakeblackshear/frigate/pull/1927#issuecomment-1003854917), but after downloading a couple of softwares for testing some ONVIF capabilities of my doorbell, I found something that was like Magic:

[Happytime ONVIF Client](http://happytimesoft.com/products/onvif-client/index.html)

With this tool, I could change the audio encoding codec that the doorbell uses from the weird PCM S16BE (which is not supported by the WebRTC card by the way) to AAC or a more standard G711. I could even change the codecs for the different channels, so I did:

Channel 1: H264 720P video with G711 audio for use with WebRTC card (for real-time audio and video)

Channel 2: H264 1080P video with AAC audio for use with Frigate (which will save me some CPU cycles because I don't have to transcode the audio codec)

https://user-images.githubusercontent.com/29582865/147899534-3cce7c4b-1e63-47cf-835b-49630009f383.mp4

PS: I could not find a way to switch back to the original codec, but I don't mind. I suppose a firmware reset would work, but didn't try.

Previous:

![image](https://user-images.githubusercontent.com/29582865/147899270-8ec00874-a957-4965-9997-e044da60f655.png)

After (disregard video differences):

![image](https://user-images.githubusercontent.com/29582865/147899115-f06e101f-7508-4bb8-8535-59f847407cf3.png)

I just wanted to document this, perhaps it helps someone in the future.

**Comment by lordCONAN** [comment]
I gave this a try, and it worked fine for a few days, but then, for seemingly no reason, the audio returned to the original codec.

**Comment by luzik** [comment]
Thanks for a hint ! This working with vto-2211g-wp

**Comment by epicurean70** [comment]
Hi, Can you share your working config for the vto221g-wp ?

much thanks!

**Comment by felipecrs** [comment]
This method has one problem, which is that it the VTO will automatically switch back to the original codec sometimes, and I don't know why. There's a better method, but it will require go2rtc:

- https://github.com/AlexxIT/go2rtc/issues/49#issuecomment-1288196783

**Comment by felipecrs** [comment]
If anyone else stumbles upon this thread, I recommend checking out https://github.com/felipecrs/dahua-vto-on-home-assistant.
